### PR TITLE
Make openTreeItemInEditor asynchronous and testable, fixes #94

### LINKED ISSUE #94: Find out the way to test vscode.window object to cover openTextDocument

### PR DESCRIPTION
**Pull request by odockal**
Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

File: test/editor/knativeOpenTextDocument.test.ts [hunk 1]
```diff
@@ -3,11 +3,15 @@ import * as chai from 'chai';
...
 import * as sinon from 'sinon';
 import * as yaml from 'yaml';
+import { expect } from 'chai';
+import { fail } from 'assert';
 import { ContextType } from '../../src/cli/config';
 import { openTreeItemInEditor } from '../../src/editor/knativeOpenTextDocument';
 import { KnativeTreeItem } from '../../src/tree/knativeTreeItem';
 import { Service } from '../../src/knative/service';
 
+import virtualfs = require('../../src/cli/virtualfs');
```
Comment (joshuawilson) [inline test/editor/knativeOpenTextDocument.test.ts:13]: Why did you use `require`?

File: test/editor/knativeOpenTextDocument.test.ts [hunk 2]
```diff
@@ -3,11 +3,15 @@ import * as chai from 'chai';
 import * as sinonChai from 'sinon-chai';
 import * as sinon from 'sinon';
 import * as yaml from 'yaml';
+import { expect } from 'chai';
```
Comment (joshuawilson) [inline test/editor/knativeOpenTextDocument.test.ts:6]: `chai` is already imported above, why not use that one?

File: test/editor/knativeOpenTextDocument.test.ts [hunk 3]
```diff
@@ -161,37 +167,70 @@ status:
...
-    // const spyShowTextDoc = sandbox.spy(stubShowTextDoc);
-    openTreeItemInEditor(testServiceTreeItem, 'yaml', false);
-    // sinon.assert.calledOnce(spyShowTextDoc);
+  test('should open a Service tree item in the editor', async () => {
+    // help stub to verify that we called virtualfs.vfsUri
+    const vfsUriStub = sandbox.stub(virtualfs, 'vfsUri');
+    vfsUriStub.returns(uriObject);
+    // creates a stub that resolves to mock text doc object, and we will spy on
+    const stubOpenTextDoc = sandbox.stub(vscode.workspace, 'openTextDocument');
+    stubOpenTextDoc.resolves(textDocumentObject);
```
Comment (joshuawilson) [inline test/editor/knativeOpenTextDocument.test.ts:176]: You can actually perform the `resolve` on the line above.

**Comment by odockal** [comment]
@joshuawilson I would like to address your remarkes but PR is already merged, should I implement these in other PR, or?

**Comment by joshuawilson** [comment]
I thought you were on PTO and so merged it and then made a commit with the changes. However, I did something wrong and could use your help looking at what is is in `main`. The UI tests are failing.

**Pull request merged**